### PR TITLE
refactor: improve api doc

### PR DESCRIPTION
**Pull request by andylokandy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,40 +16,58 @@
...
-Add the dependencies to your `Cargo.toml` with:
+## Getting Started
+
+Add `log` and `logforth` to your `Cargo.toml`:
 
 ```shell
-cargo add log
-cargo add logforth
+> cargo add log
+> cargo add logforth
```
Comment (tisonkun) [inline README.md:34]:
```suggestion
cargo add log
cargo add logforth
```

File: examples/custom_layout_filter.rs [hunk 2]
```diff
@@ -21,24 +20,22 @@ use logforth::layout::CustomLayout;
...
 
-    log::error!("Hello error!");
-    log::warn!("Hello warn!");
-    log::info!("Hello info!");
-    log::debug!("Hello debug!");
-    log::trace!("Hello trace!");
+    log::error!("This is an error.");
+    log::warn!("This is a warning.");
+    log::info!("This info will not be logged.");
 }
```
Comment (tisonkun) [inline examples/custom_layout_filter.rs:41]: I'd prefer to keep the original log demo lines. Ditto other examples.

File: examples/rolling_file.rs [hunk 3]
```diff
@@ -16,39 +16,23 @@ use logforth::append::rolling_file::NonBlockingBuilder;
...
-        .rotation(Rotation::Minutely)
-        .filename_prefix("example")
-        .filename_suffix("log")
-        .max_log_files(10)
-        .max_file_size(1024 * 1024)
+    let rolling_writer = RollingFileWriter::builder()
+        .rotation(Rotation::Daily)
+        .filename_prefix("app_log")
         .build("logs")
         .unwrap();
```
Comment (tisonkun) [inline examples/rolling_file.rs:26]: Revert these unrelated changes.

File: src/append/mod.rs [hunk 4]
```diff
@@ -12,10 +12,18 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Dispatch log records to the appropriate target.
+//! Various appenders for log records.
```
Comment (tisonkun) [inline src/append/mod.rs:15]:
```suggestion
//! Dispatch log records to the appropriate target.
```

Keep the original doc. It's more appropriate to me.

File: src/append/mod.rs [hunk 5]
```diff
@@ -25,16 +33,11 @@ pub use self::rolling_file::RollingFile;
...
-#[cfg(feature = "rolling_file")]
-pub mod rolling_file;
-mod stdio;
-
+/// A trait representing an appender that can process log records.
+///
+/// Implementors of this trait can handle log records in custom ways.
 pub trait Append: fmt::Debug + Send + Sync + 'static {
-    /// Dispatches a log record to the append target.
+    /// Processes a log record.
```
Comment (tisonkun) [inline src/append/mod.rs:40]:
```suggestion
    /// Dispatches a log record to the append target.
```

File: src/filter/mod.rs [hunk 6]
```diff
@@ -24,20 +24,23 @@ pub use self::env_filter::EnvFilter;
...
 pub enum FilterResult {
-    /// The record will be processed without further filtering.
+    /// Accept the log record.
     Accept,
-    /// The record should not be processed.
+    /// Reject the log record.
     Reject,
-    /// No decision could be made, further filtering should occur.
+    /// Neutral; defer to other filters.
     Neutral,
```
Comment (tisonkun) [inline src/filter/mod.rs:35]: Let's keep the original lines.

File: examples/rolling_file.rs (hunk 3 above)
Comment (andylokandy) [reply examples/rolling_file.rs:26]: This is intended to simplify the example. I think the original is overcomplicated and may lead the reader to confuse.

Comment (tisonkun) [reply examples/rolling_file.rs:26]: Oops. This is in the example dir. Then it's fine.

File: src/append/opentelemetry.rs [hunk 7]
```diff
@@ -32,10 +34,7 @@ use opentelemetry_sdk::logs::LoggerProvider;
 use crate::append::Append;
 use crate::Layout;
 
-/// The communication protocol to opentelemetry that used when exporting data.
-///
-/// This is a logical re-exported [`Protocol`] to avoid version lock-in to
-/// `opentelemetry_otlp`.
+/// Specifies the wire protocol to use when sending logs to OpenTelemetry.
```
Comment (tisonkun) [inline src/append/opentelemetry.rs:37]:
```suggestion
/// Specifies the wire protocol to use when sending logs to OpenTelemetry.
///
/// This is a logical re-exported [`Protocol`] to avoid version lock-in to
/// `opentelemetry_otlp`.
```

**Review by tisonkun — APPROVED** [review]
Rest LGTM.

**Comment by tisonkun** [comment]
Let me release a 0.15.1 for this. We don't introduce any API changes, do we?

File: src/layout/json.rs [hunk 8]
```diff
@@ -36,11 +36,33 @@ use crate::layout::Layout;
...
+///
+/// ```
+/// use logforth::layout::JsonLayout;
+///
+/// let json_layout = JsonLayout::default();
+/// ```
 #[derive(Default, Debug, Clone)]
 pub struct JsonLayout {
-    pub tz: Option<TimeZone>,
+    tz: Option<TimeZone>,
```
Comment (tisonkun) [inline src/layout/json.rs:48]: Well. This is one. Then let me release a 0.16.0
